contracts/strategies/harvest/IdleHarvestStrategy.sol

```diff
@@ -0,0 +1,270 @@
+        govTokens[0] = govToken;
+        return govTokens;
+    }
+
+    function getApr() external view returns (uint256) {
+        uint256 rawBalance = IRewardPool(rewardPool).balanceOf(address(this));
+        uint256 expectedUnderlyingAmount = (price() * rawBalance) / oneToken;
+
+        if (expectedUnderlyingAmount <= lastIndexAmount) {
+            return 0;
+        }
+
+        uint256 gain = expectedUnderlyingAmount - lastIndexAmount;
+        uint256 time = block.timestamp - lastIndexedTime;
+        uint256 gainPerc = (gain * 10**20) / lastIndexAmount;
+        uint256 apr = (YEAR / time) * gainPerc;
+        return apr;
+    }
+
+    /// @notice Redeem Tokens
+    /// @param _amount amount of strategy tokens to redeem
+    /// @return Amount of underlying tokens received
+    function redeem(uint256 _amount) external override onlyIdleCDO returns (uint256) {
+        if (_amount > 0) {
+            return _redeem(_amount);
+        }
+    }
+
+    /// @notice Redeem Tokens
+    /// @param _amount amount of underlying tokens to redeem
+    /// @return Amount of underlying tokens received
+    function redeemUnderlying(uint256 _amount) external returns (uint256) {
+        if (_amount > 0) {
+            uint256 _underlyingAmount = (_amount * oneToken) / price();
+            return _redeem(_underlyingAmount);
+        }
+    }
+
+    /// @notice Internal function to redeem the underlying tokens
+    /// @param _amount Amount of strategy tokens
+    /// @return Amount of underlying tokens received
+    function _redeem(uint256 _amount) internal returns (uint256) {
+        lastIndexAmount = lastIndexAmount - _amount;
+        lastIndexedTime = block.timestamp;
+        _burn(msg.sender, _amount);
+        IRewardPool(rewardPool).withdraw(_amount);
+        IERC20Detailed _underlyingToken = underlyingToken;
+        uint256 balanceBefore = _underlyingToken.balanceOf(address(this));
+        IHarvestVault(strategyToken).withdraw(_amount);
+        uint256 balanceAfter = _underlyingToken.balanceOf(address(this));
+        uint256 balanceReceived = balanceAfter - balanceBefore;
+        _underlyingToken.safeTransfer(msg.sender, balanceReceived);
+        return balanceReceived;
+    }
+
+    /// @notice Deposit the underlying token to vault
+    /// @param _amount number of tokens to deposit
+    /// @return minted number of reward tokens minted
+    function deposit(uint256 _amount) external override onlyIdleCDO returns (uint256 minted) {
+        if (_amount > 0) {
+            underlyingToken.safeTransferFrom(msg.sender, address(this), _amount);
+            lastIndexAmount = lastIndexAmount + _amount;
+            minted = _depositToVault(_amount);
+        }
+    }
+
+    /// @notice internal function to deposit the funds to the vault
+    /// @param _amount Amount of tokens to deposit
+    function _depositToVault(uint256 _amount) internal returns (uint256) {
+        underlyingToken.safeApprove(strategyToken, _amount);
```

Comment:
should avoid reading a storage variable repeately.
`address _strategyToken = strategyToken;`

Reply:
+1

Reply:
this is not resolved